url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/GadajAsterisk.java

```diff
@@ -0,0 +1,695 @@
+import pl.cc.exceptions.AgentNotFoundException;
+import pl.cc.real.RealAgent;
+import pl.cc.real.RealAgentList;
+import pl.cc.real.RealCall;
+import pl.cc.real.RealQueue;
+
+public class GadajAsterisk extends Thread {
+	static Logger logMessageFromAsterisk = Logger.getLogger("logMessageFromAsterisk");
+	static Logger log = Logger.getLogger(GadajAsterisk.class);
+
+	static String asteriskPort;
+	static int numer = 0;
+	static String asteriskIP;
+	static String asteriskUser;
+	static String asteriskPass;
+
+	static Socket echoSocket = null;
+	static PrintWriter out = null;
+	static BufferedReader in = null;
+	boolean asteriskReader = true;
+	static boolean connected = true;
+
+	static Parametry p;
+
+	static WiadomoscOdAgentLlist wiadomosciOdAgenta = new WiadomoscOdAgentLlist();
+	static WiadomoscOdAsteriskaList wiadomosciOdAsteriska = new WiadomoscOdAsteriskaList();
+
+	static ReentrantLock agentListLock= new ReentrantLock(true);
+
+	public GadajAsterisk() {
+        asteriskReader=false;
+	}
+
+	public GadajAsterisk(String asteriskIP, String asteriskPort,
+			String asteriskUser, String asteriskPass, Parametry parametry)
+			throws Exception {
+		super();
+		this.p = parametry;
+		this.asteriskPort = asteriskPort;
+		this.asteriskIP = asteriskIP;
+		this.asteriskUser = asteriskUser;
+		this.asteriskPass = asteriskPass;
+
+		try {
+			if (echoSocket == null) {
+				echoSocket = new Socket(asteriskIP, new Integer(asteriskPort)
+						.intValue());
+				out = new PrintWriter(echoSocket.getOutputStream());
+				in = new BufferedReader(new InputStreamReader(echoSocket
+						.getInputStream()));
+
+				// Autoryzacja do asteriska
+				if (!authAsterisk())
+					throw new Exception();
+				// Listujemy kolejki wszystkie i zbieramy info
+				sendAction("queuestatus");
+			}
+		} catch (UnknownHostException e) {
+			System.err.println("Problem z połączeniem do Asteriska");
+			System.exit(1);
+		} catch (IOException e) {
+			System.err.println("Problem z połączeniem do Asteriska");
+			System.exit(1);
+		}
+	}
+
+	public void close() {
+		try {
+			echoSocket.close();
+			asteriskPort = null;
+			numer = 0;
+			asteriskIP = null;
+			asteriskUser = null;
+			asteriskPass = null;
+
+			echoSocket = null;
+			out = null;
+			in = null;
+			asteriskReader = true;
+			connected = true;
+
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+
+	public void sendAction(String action) {
+		out.print("Action: " + action + "\r\n\r\n");
+		out.flush();
+	}
+
+	private boolean authAsterisk() throws IOException {
+		String line = in.readLine();
+		System.out.println(line);
+
+		out.print("Action: login\r\n");
+		out.print("username: " + asteriskUser + "\r\n");
+		out.print("secret: " + asteriskPass + "\r\n");
+		out.print("\r\n");
+		out.flush();
+		line = in.readLine();
+		System.out.println(line);
+		if (line.equals("Response: Success")) {
+			return true;
+		}
+
+		return false;
+	}
+
+	public void run() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.